lib/diagnostics_channel.js

```diff
@@ -95,59 +76,89 @@ function wrapStoreRun(store, data, next, transform = defaultTransform) {
   };
 }
 
-// TODO(qard): should there be a C++ channel interface?
-class ActiveChannel {
+const channels = new WeakRefMap();
+
+class Channel {
+  #subscribers;
+  #stores;
+  #active = false;
+
+  constructor(name) {
+    this.name = name;
+    channels.set(name, this);
+  }
+
+  get hasSubscribers() {
+    return this.#active;
+  }
+
+  #ensureActive() {
+    if (!this.#active) {
+      this.#active = true;
+      this.#subscribers = [];
+      this.#stores = new SafeMap();
+    }
+  }
+
+  #maybeDeactivate() {
+    if (this.#active && !this.#subscribers.length && !this.#stores.size) {
+      this.#active = false;
+      this.#subscribers = undefined;
```

Comment:
This changes `this._subscribers` to `this.#subscribers` which is a breaking change.